docs/guides/python/serverless-ai-api.mdx

```diff
@@ -0,0 +1,348 @@
+- We then call the OpenAI API (using the ChatGPT model `gpt-3.5-turbo`) with a prompt asking to summarize the input text. We limit the `max_tokens` for the summary to ~100 tokens (roughly a few sentences) to keep it concise.
+- The OpenAI response contains a list of choices; we extract the content of the first choice (which is the model's summary text).
+- We set `ctx.res.body` to a JSON object containing the summary. Nitric will serialize this to JSON for the HTTP response. By default, returning a Python dict as `res.body` results in a JSON response.
+- If the OpenAI API call fails (e.g., network issue or invalid key), we catch the exception and return a 500 error with the message.
+
+_A note on async:_ We defined the handler as `async def`. The Nitric framework supports async handlers, which is great for I/O operations. The OpenAI Python SDK call we used (`ChatCompletion.create`) is a blocking call. In a real app, you might integrate an async HTTP client or use OpenAI's async methods (if available) to avoid blocking the event loop. However, for simplicity in this tutorial, this is acceptable - our focus is on the integration, not optimizing async usage.
+
+5. **Finalize the Nitric app run:** After defining all your routes and handlers in the file, make sure to add:
+
+```python
+Nitric.run()
+```
+
+This call tells Nitric to start the application (it will scan for all defined services/routes and run the local server when invoked via `nitric start`). In our case, it picks up `summarize_api` and the attached route.
+
+That's it for our service code! We have a complete Python backend that will accept text and return an AI-generated summary. The entire `summarize.py` file should look like this (for reference):
+
+```python {{ title: services/summarize.py }}
+import os
+from openai import OpenAI
+from nitric.resources import api
+from nitric.application import Nitric
+from nitric.context import HttpContext
+
+# Configure OpenAI API key from env
+api_key = os.environ.get("OPENAI_API_KEY")
+if not api_key:
+  raise Exception("Please set the OPENAI_API_KEY environment variable")
+
+# Create OpenAI client
+client = OpenAI(api_key=api_key)
+
+# Create a Nitric API named 'main'
+summarize_api = api("main")
+
+@summarize_api.post("/summarize")
+async def summarize_text(ctx: HttpContext) -> None:
+  # Extract input text from request (assuming JSON body with a 'text' field)
+  req_data = ctx.req.json
+  if not req_data or 'text' not in req_data:
+    ctx.res.status = 400
+    ctx.res.body = {"error": "No text provided for summarization."}
+    return
+
+  text_to_summarize = req_data['text']
+  # Call OpenAI API to get summary (this is a blocking call, for demo simplicity)
+  try:
+    response = client.chat.completions.create(
+      model="gpt-3.5-turbo",
```

Comment:
should we allow this be to be configurable?

Reply:
suggest we either add a note or extend this as a future improvement.